lodash.js

```diff
@@ -10424,6 +10424,41 @@
     }
 
     /**
+     * Converts a `value` to an array.
+     *
+     * @static
+     * @memberOf _
+     * @category Lang
+     * @param {*} value The value to convert.
+     * @returns {Array} Returns the converted array.
+     * @example
+     *
+     * _.toArray({ 'a': 1, 'b': 2 });
+     * // => [{ 'a': 1, 'b': 2 }]
+     *
+     * _.toArray([1, 2, 3]);
+     * // => [1, 2, 3]
+     *
+     * _.toArray('abc');
+     * // => ['abc']
+     *
+     * _.toArray(1);
+     * // => [1]
+     *
+     * _.toArray(null);
+     * // => [null]
+     */
+    function asArray(value) {
+      if (isArray(value)) {
+        return copyArray(value);
+      }
+      if (iteratorSymbol && value && value[iteratorSymbol] && !isString(value)) {
+        return iteratorToArray(value[iteratorSymbol]());
+      }
+      return [value];
+    }
+
+    /**
```

Comment:
So no clone in this PR?
On Feb 12, 2016 4:37 PM, "John-David Dalton" <notifications@github.com>
wrote:

> In lodash.js
> <https://github.com/lodash/lodash/pull/1993#discussion_r52801733>:
>
> > +     * // => [1]
> > +     *
> > +     * _.toArray(null);
> > +     * // => [null]
> > +     */
> > +    function asArray(value) {
> > +      if (isArray(value)) {
> > +        return copyArray(value);
> > +      }
> > +      if (iteratorSymbol && value && value[iteratorSymbol] && !isString(value)) {
> > +        return iteratorToArray(value[iteratorSymbol]());
> > +      }
> > +      return [value];
> > +    }
> > +
> > +    /**
>
> Cool. Adding shallow cloning is easy enough before we bump. I'll ask for
> feedback on the tweets.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lodash/lodash/pull/1993/files#r52801733>.
>